pkg/queryfrontend/request.go

```diff
@@ -114,8 +114,9 @@ func (r *ThanosLabelsRequest) GetStart() int64 { return r.Start }
 // GetEnd returns the end timestamp of the request in milliseconds.
 func (r *ThanosLabelsRequest) GetEnd() int64 { return r.End }
 
-// GetStep returns the step of the request in milliseconds.
-func (r *ThanosLabelsRequest) GetStep() int64 { return 0 }
+// GetStep returns the step of the request in milliseconds. Returns 1 is a trick to avoid panic in
+// https://github.com/cortexproject/cortex/blob/master/pkg/querier/queryrange/results_cache.go#L447.
```

Comment:
(: Yea it's getting very tricky without Cortex actually support our case, and it would be nice to help them as well :thinking: 
This is fine for now, but wonder how long term solution of truly shared code with cortex would looks like

Reply:
Yeah, I will take a look later. Would be good to have these query APIs natively supported on Cortex then we can just re-use.